qiskit/primitives/containers/shape.py

```diff
@@ -60,14 +60,17 @@ def __repr__(self):
 
     @property
     def shape(self):
+        """shape (tuple[int]): The leading shape of every entry in the instance."""
         return self._shape
 
     @property
     def ndim(self):
+        """ndim (int): The length of the shape attribute."""
         return len(self._shape)
 
     @property
     def size(self):
+        """size (int): The product of the entries in the shape attribute."""
         return int(np.prod(self._shape, dtype=int))
```

Comment:
This inherits from `Shaped` and can safely be typed as such.

```suggestion
    @property
    def shape(self) -> tuple[int, ...]:
        """The array shape of this object."""
        return self._shape

    @property
    def ndim(self) -> int:
        """The length of the shape attribute."""
        return len(self._shape)

    @property
    def size(self) -> int:
        """The product of the entries in the shape attribute."""
        return int(np.prod(self._shape, dtype=int))
```

Because of the inheritance, do we actually need docstrings?

```suggestion
    @property
    def shape(self) -> tuple[int, ...]:
        return self._shape

    @property
    def ndim(self) -> int:
        return len(self._shape)

    @property
    def size(self) -> int:
        return int(np.prod(self._shape, dtype=int))
```